uk.co.compucorp.civicrm.hrcore/CRM/HRCore/Upgrader/Steps/1017.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+trait CRM_HRCore_Upgrader_Steps_1017 {
+
+  /**
+   * Migrating Phone Types and Deleting unwanted Phone Types
+   */
+  public function upgrade_1017() {
+    $this->migratePhoneTypes();
+    $this->relabelPhoneType("phone", "Landline");
+    $this->reservePhoneTypes(["phone", "mobile"]);
+    $this->deletePhoneTypes(["fax", "pager", "voicemail"]);
+    return TRUE;
+  }
+
+
+  /**
+   * Migrate the Phone Type Data as Follows:
+   * Fax -> LandLine
+   * Pager ->Mobile
+   * Voicemai -> Mobile
+   */
+  private function migratePhoneTypes() {
+    $phoneTypes = [
+      'Fax' => 'Phone',
+      'Pager' => 'Mobile',
+      'Voicemail' => 'Mobile',
+    ];
+    foreach ($phoneTypes as $originalType => $newType) {
+      $result = civicrm_api3('Phone', 'get', [
+        'sequential' => 1,
+        'return' => ["id", "phone_type_id", "is_primary", "phone"],
+        'phone_type_id' => $originalType,
+      ]);
+      if ($result['count'] == 0) {
+        continue;
+      }
+
+      foreach ($result['values'] as $phone) {
+        $result = civicrm_api3('Phone', 'create', [
+          'id' => $phone['id'],
+          'phone_type_id' => $newType,
+          'is_primary' => $phone['is_primary'],
```

Comment:
Do we need this `'is_primary' => $phone['is_primary'],` ? Since what we need to update the `phone_type_id` field is just the phone ID. The is_primary should remain the same since we are not modifying that.